### PR TITLE
Pdf refactor

### PR DESCRIPTION
**Pull request by minottic**
Draft PR to collect feedbacks. 

Tests are missing and style might be improved

### CODE REVIEW

File: sci-log-db/Dockerfile [hunk 1]
```diff
@@ -1,9 +1,13 @@
 # Check out https://hub.docker.com/_/node to select a new base image
-FROM node:14-slim
+FROM node:18-slim
 
 # Install extra packages
 RUN apt-get update && apt-get install -y zip \
- texlive-latex-extra
+ texlive-latex-extra && \
```
Comment (wakonig) [inline sci-log-db/Dockerfile:6]: I guess latex is not needed anymore?

File: sci-log-db/src/services/export-snippets.service.ts [hunk 2]
```diff
@@ -1,543 +1,142 @@
...
-          if (error) {
-            console.warn(error);
-          }
-
-          resolve(this.exportDir + '/all.zip');
-        },
-      );
-    });
+  private quote = (snippet: Paragraph, element: Element) => {
+    if (snippet.linkType === 'quote') {
```
Comment (wakonig) [inline sci-log-db/src/services/export-snippets.service.ts:37]:
```
if (snippet.linkType !== 'quote'){
    return;
}
...
```

might be a bit more concise

**Review by wakonig — APPROVED** [review]
Now that's what I call a cleanup! Great job and thanks a lot, Carlo!

File: sci-log-db/Dockerfile (hunk 1 above)
Comment (minottic) [reply sci-log-db/Dockerfile:6]: correct, thanks for noticing that! I will remove it in the next commit

File: sci-log-db/src/services/export-snippets.service.ts (hunk 2 above)
Comment (minottic) [reply sci-log-db/src/services/export-snippets.service.ts:37]: ok, I'll change to this everywhere

File: sci-log-db/src/mixins/basesnippet.repository-mixin.ts [hunk 3]
```diff
@@ -310,29 +311,46 @@ function ExportRepositoryMixin<
   R extends MixinTarget<DefaultCrudRepository<M, ID, Relations>>,
 >(superClass: R) {
   class Mixed extends superClass {
-    exportDir: string;
     @inject.getter(EXPORT_SERVICE)
     readonly exportServiceGetter: Getter<ExportService>;
 
+    private exportFile = 'export.pd'
+    private basePath = 'tmp'
```
Comment (minottic) [inline sci-log-db/src/mixins/basesnippet.repository-mixin.ts:318]: notetoself: remember to change this to /tmp

File: sci-log-db/src/mixins/basesnippet.repository-mixin.ts [hunk 4]
```diff
@@ -310,29 +311,46 @@ function ExportRepositoryMixin<
   R extends MixinTarget<DefaultCrudRepository<M, ID, Relations>>,
 >(superClass: R) {
   class Mixed extends superClass {
-    exportDir: string;
     @inject.getter(EXPORT_SERVICE)
     readonly exportServiceGetter: Getter<ExportService>;
 
+    private exportFile = 'export.pd'
```
Comment (minottic) [inline sci-log-db/src/mixins/basesnippet.repository-mixin.ts:317]: notetoself: remember to change this to .pdf

**Pull request merged**